os/drivers/usbhost/usbhost_enumerate.c

```diff
@@ -467,8 +468,19 @@ int usbhost_enumerate(FAR struct usbhost_hubport_s *hport, FAR struct usbhost_cl
 
 	if (cfglen > maxlen) {
 		udbg("ERROR: Configuration doesn't fit in buffer, length=%d, maxlen=%d\n", cfglen, maxlen);
-		ret = -E2BIG;
-		goto errout;
+		/* Most of the USB class configuration descriptor would be with 128bytes limit,
+		 * but in case of uvc class the size of the configuration decriptor would be in the
+		 * range of 1000 - 2000bytes. Hence it is required to reallocate the initial buffer
+		 * used to read size of the configuration descriptor
+		 */
+		DRVR_FREE(hport->drvr, buffer);
+		buffer = NULL;
+		ret = DRVR_IOALLOC(hport->drvr, &buffer, cfglen);
```

Comment:
Should we maintain allocated memory in line 326?

Reply:
Yes, In USB when we read the configuration descriptor for the first time we dont know the size of configuration descriptor. As per the USB spec it should read the first 8/64 bytes(FULL Speed/HIGH Speed) and get the actual of configuration descriptor wTotalLength variable. With this new size we have allocate another buffer and read it. 

As you can see in the patch at line 476 i release the memory allocated at 376 and allocate a bigger buffer.

DRVR_ALLOC can return only a fixed memory size of 128bytes, which is enough in most the devices except for UVC class.

DRVR_IOALLOC support memory with variable size.

Reply:
ah, sorry I missed DRVR_FREE in line 476.